File: Problems/No.231_Power_of_Two/power_of_two.py
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """

        # check if it is negative or not
        if (n < 0):
            return False

        # convert into bitset and check if the number of 1 bit is only 1
        Tbits = format(n, "032b")
        return True if (Tbits.count("1") == 1) else False

File: Problems/No.231_Power_of_Two/power_of_two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        
        // check if it is negative or not
        if (n < 0) return false;

        // convert into bitset and check if the number of 1 bit is only 1
        std::bitset<32> Tbits(n);
        return true ? (Tbits.count() == 1) : false;

    }
};
